demo/site/src/util/graphQLClient.ts

```diff
@@ -5,19 +5,33 @@ import {
     SitePreviewData,
 } from "@comet/cms-site";
 
-export function createGraphQLFetch(previewData?: SitePreviewData) {
+import { getVisibilityParam } from "./ServerContext";
+
+export function createGraphQLFetch() {
     if (typeof window !== "undefined") {
         throw new Error("createGraphQLFetch: cannot use on client side.");
     }
+    if (process.env.NEXT_RUNTIME === "edge") {
+        throw new Error("createGraphQLFetch: cannot use in edge runtime, use createGraphQLFetchMiddleware instead.");
+    }
```

Comment:
Wouldn't it be possible to move the `getVisibilityParam()` call into this if an use the same GraphQL fetch in the middleware and pages?